middle_end/flambda2/kinds/flambda_kind.ml

```diff
@@ -163,6 +173,95 @@ let is_naked_float t =
   | Region | Rec_info ->
     false
 
+module Mixed_block_shape = struct
+  type kind = t
+
+  type t =
+    { fields : kind array;
```

Comment:
I don't know if this would be the right place, but if this where to have subkinds at some point, we would need to update CSE.